src/canvasObject.js

```diff
@@ -52,6 +53,7 @@ CanvasObject.prototype = {
             if(previousImageObj){
               viewer.world.removeItem(previousImageObj);
             }
+            document.dispatchEvent(new CustomEvent('detail-tile-source-opened', { 'detail': self.id }));
```

Comment:
Thanks- that branch had slipped my mind when I was working on this, so I'll take a look at it as a reference, and move forward from there.